probe-rs/src/probe/cmsisdap/commands/transfer/mod.rs

```diff
@@ -124,10 +160,10 @@ impl Request for TransferRequest {
         Ok(size)
     }
 
-    fn from_bytes(&self, buffer: &[u8]) -> Result<Self::Response, SendError> {
+    fn from_bytes(&self, mut buffer: &[u8]) -> Result<Self::Response, SendError> {
```

Comment:
It's so that you can do `buffer = &buffer[len..];` to remove the already-processed bytes. Note that this doesn't let you mutate the pointed-to bytes, just mutate the local var. I can change it to use offsets if you think it's clearer.

Reply:
I think it was too early when I read this, just making the variable `mut` is fine. I thought it was `&mut [u8]`.